File: DemoLogin/DemoAuthFirebaseJava/app/src/main/java/com/example/muhammadfarhan/demoauthfirebasejava/MainActivity.java
package com.example.muhammadfarhan.demoauthfirebasejava;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;


// Parent Class which Hold all Fragments
public class MainActivity extends AppCompatActivity {

    private twitterInterFace twitterInterFace;

    public void setTwitterInterFace(twitterInterFace twitterInterFace) {
        this.twitterInterFace = twitterInterFace;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (twitterInterFace != null)
            twitterInterFace.myOnActivityResult(requestCode, resultCode, data);
    }

    public interface twitterInterFace {
        void myOnActivityResult(int requestCode, int resultCode, @Nullable Intent data);
    }
}

File: DemoLogin/DemoAuthFirebaseKotlin/app/src/main/java/com/example/muhammadfarhan/demoauthfirebasekotlin/MainActivity.kt
package com.example.muhammadfarhan.demoauthfirebasekotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle


class MainActivity : AppCompatActivity() {

    private var mTwitterInterFace: TwitterInterFace? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (mTwitterInterFace != null)
            mTwitterInterFace?.myOnActivityResult(requestCode, resultCode, data)
    }

    fun setTwitterInterFace(twitterInterFace: TwitterInterFace) {
        this.mTwitterInterFace = twitterInterFace
    }

    interface TwitterInterFace {
        fun myOnActivityResult(requestCode: Int, resultCode: Int, data: Intent?)
    }
}
